OpenRA.Mods.Common/Traits/GainsExperience.cs

```diff
@@ -19,7 +19,7 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("This actor's experience increases when it has killed a GivesExperience actor.")]
 	public class GainsExperienceInfo : ITraitInfo, Requires<ValuedInfo>, Requires<UpgradeManagerInfo>
 	{
-		[FieldLoader.Require]
+		[FieldLoader.Require, UpgradeGrantedReference, UpgradeUsedReference] // HACK: Treat upgrades as used
```

Comment:
Added `UpgradeUsedReference` as a hack to silence:
`OpenRA.Utility(1,1): Warning: Actor type `<insert name here>` grants upgrade `eliteweapon` that is not used by anything!`

Another option would be to add an `IgnoreUsedUpgrades` trait that pretends to use upgrades to silence warning about them being unused.